### PR TITLE
Update FB internal dependencies

### PR DESCRIPTION
**Pull request by ajkr**
I noticed we were building against zstd 1.3.0 which is missing optimizations that our fbcode customers have (they're on zstd 1.3.5). Ran `./build_tools/update_dependencies.sh` to catch us up. Omitted the changes it made for gcc-4.8 since it's broken.

Test Plan:

- `make check -j64`

### CODE REVIEW

File: build_tools/dependencies.sh [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
+BZIP2_BASE=/mnt/gvfs/third-party2/bzip2/7efd501e9b339c310ffe42a0fe502e76a139153e/1.0.6/gcc-5-glibc-2.23/9bc6787
+LZ4_BASE=/mnt/gvfs/third-party2/lz4/907b498203d297947f3bb70b9466f47e100f1873/r131/gcc-5-glibc-2.23/9bc6787
+ZSTD_BASE=/mnt/gvfs/third-party2/zstd/be97d82d5f8117a4267d1257ea8a452b6eafc7a8/1.3.5/gcc-5-glibc-2.23/03859b5
+GFLAGS_BASE=/mnt/gvfs/third-party2/gflags/0b9929d2588991c65a57168bf88aff2db87c5d48/2.2.0/gcc-5-glibc-2.23/9bc6787
+JEMALLOC_BASE=/mnt/gvfs/third-party2/jemalloc/da5d5bd759f4b19de1f9d54dd0041f9af9f3244f/master/gcc-5-glibc-2.23/aa64d6b
+NUMA_BASE=/mnt/gvfs/third-party2/numa/9cbf2460284c669ed19c3ccb200a71f7dd7e53c7/2.0.11/gcc-5-glibc-2.23/9bc6787
+LIBUNWIND_BASE=/mnt/gvfs/third-party2/libunwind/bf3d7497fe4e6d007354f0adffa16ce3003f8338/1.3/gcc-5-glibc-2.23/b443de1
+TBB_BASE=/mnt/gvfs/third-party2/tbb/ff4e0b093534704d8abab678a4fd7f5ea7b094c7/2018_U5/gcc-5-glibc-2.23/9bc6787
+KERNEL_HEADERS_BASE=/mnt/gvfs/third-party2/kernel-headers/b5c4a61a5c483ba24722005ae07895971a2ac707/4.0.9-36_fbk5_2933_gd092e3f/gcc-5-glibc-2.23/da39a3e
+BINUTILS_BASE=
```
Comment (siying) [inline build_tools/dependencies.sh]: Is it OK?

Comment (ajkr) [reply build_tools/dependencies.sh]: Idk, we can try upgrading centos6-native to centos7-native and see what happens. These packages for centos6-native aren't listed in third-party2 anymore.

**Comment by yiwu-arbug** [comment]
How about we update the file as an extra step before every time we release to fbcode?

**Comment by ajkr** [comment]
@yiwu-arbug Sounds good, we can do it when we're bumping version number in version.h and HISTORY.md.

**Pull request closed without merging**